mage_ai/data_preparation/executors/pipeline_executor.py

```diff
@@ -119,19 +121,29 @@ async def execute_block() -> None:
                     execution_partition=self.execution_partition,
                 )
                 block_run_data = dict(status=BlockRun.BlockRunStatus.RUNNING)
-                if not block_run.started_at or \
-                        (block_run.metrics and not block_run.metrics.get('controller')):
+                if not block_run.started_at or (
+                    block_run.metrics and not block_run.metrics.get('controller')
+                ):
                     block_run_data['started_at'] = datetime.now(tz=pytz.UTC)
 
                 block_run.update(**block_run_data)
 
                 try:
-                    return BlockExecutor(block_run_id=block_run.id, **executor_kwargs).execute(
+                    # Get retry_config from block only (open source version)
+                    block = self.pipeline.get_block(block_run.block_uuid)
+                    retry_config = (
+                        block.retry_config if block and block.retry_config else None
+                    )
+
+                    return BlockExecutor(
+                        block_run_id=block_run.id, **executor_kwargs
+                    ).execute(
                         block_run_id=block_run.id,
                         block_run_outputs_cache=block_run_outputs_cache,
                         cache_block_output_in_memory=self.pipeline.cache_block_output_in_memory,
                         global_vars=global_vars,
                         pipeline_run_id=pipeline_run.id,
+                        retry_config=retry_config,
```

Comment:
This is not needed. The BlockExecutor will get the retry config https://github.com/mage-ai/mage-ai/blob/master/mage_ai/data_preparation/executors/block_executor.py#L598-L601